src/components/App/NamespaceRoutes.tsx

```diff
@@ -4,24 +4,28 @@ import { useTranslation } from 'react-i18next';
 
 import { languageAtom } from 'state/preferences/languageAtom';
 import { extensionsState } from 'state/navigation/extensionsAtom';
-import { namespacesState } from 'state/namespacesAtom';
 
 import { resourceRoutesNamespaced } from 'resources';
 import { createExtensibilityRoutes } from './ExtensibilityRoutes';
 import { otherRoutesNamespaced } from 'resources/other';
-
 import { IncorrectPath } from './IncorrectPath';
 import { useUrl } from 'hooks/useUrl';
+import { useGet } from 'shared/hooks/BackendAPI/useGet';
 
 export default function NamespaceRoutes() {
   const { t } = useTranslation();
   const { namespaceId } = useParams();
   const { clusterUrl } = useUrl();
   const language = useRecoilValue(languageAtom);
   const extensions = useRecoilValue(extensionsState);
-  const namespaces = useRecoilValue(namespacesState);
 
-  if (!namespaces.includes(namespaceId as string))
+  const { error } = useGet(`/api/v1/namespaces/${namespaceId}`, {
+    skip: false,
+    pollingInterval: 0,
+    onDataReceived: () => {},
+  });
+
+  if (error)
```

Comment:
if the contents of an if don't fit in a single line i would suggest going with 
```suggestion
  if (error) {
```
instead